src/Generator/ClassGenerator.php

```diff
@@ -448,8 +465,28 @@ public function getImplementedInterfaces()
     }
 
     /**
+     * @param string $implementedInterface
+     * @return bool
+     */
+    public function hasImplementedInterface($implementedInterface)
+    {
+        return in_array($implementedInterface, $this->implementedInterfaces);
```

Comment:
I would also check against `\\` strings. Since you are basing against the 7.0 branch, you may want to check the type generator class